typing_extensions/src_py3/test_typing_extensions.py

```diff
@@ -0,0 +1,636 @@
+        self.assertNotIsInstance(42, typing_extensions.Collection)
+
+    @skipUnless(TYPING_V2, "Behavior added in typing v2")
+    def test_collection_instantiation(self):
+        class MyCollection(typing_extensions.Collection[int]):
+            def __contains__(self, item): ...
+            def __iter__(self): ...
+            def __len__(self): ...
+
+        self.assertIsSubclass(
+            type(MyCollection()),
+            typing_extensions.Collection)
+        self.assertIsSubclass(
+            MyCollection,
+            typing_extensions.Collection)
+
+    def test_deque(self):
+        self.assertIsSubclass(collections.deque, typing_extensions.Deque)
+        class MyDeque(typing_extensions.Deque[int]): ...
+        self.assertIsInstance(MyDeque(), collections.deque)
+
+    def test_counter(self):
+        self.assertIsSubclass(collections.Counter, typing_extensions.Counter)
+
+    @skipUnless(TYPING_V4, "Behavior added in typing v4")
+    def test_defaultdict_instantiation(self):
+        self.assertIs(
+            type(typing_extensions.DefaultDict()),
+            collections.defaultdict)
+        self.assertIs(
+            type(typing_extensions.DefaultDict[KT, VT]()),
+            collections.defaultdict)
+        self.assertIs(
+            type(typing_extensions.DefaultDict[str, int]()),
+            collections.defaultdict)
+
+    def test_defaultdict_subclass(self):
+
+        class MyDefDict(typing_extensions.DefaultDict[str, int]):
+            pass
+
+        dd = MyDefDict()
+        self.assertIsInstance(dd, MyDefDict)
+
+        self.assertIsSubclass(MyDefDict, collections.defaultdict)
+        if TYPING_V3:
+            self.assertNotIsSubclass(collections.defaultdict, MyDefDict)
+
+    def test_chainmap_instantiation(self):
+        self.assertIs(type(typing_extensions.ChainMap()), collections.ChainMap)
+        self.assertIs(type(typing_extensions.ChainMap[KT, VT]()), collections.ChainMap)
+        self.assertIs(type(typing_extensions.ChainMap[str, int]()), collections.ChainMap)
+        class CM(typing_extensions.ChainMap[KT, VT]): ...
+        if TYPING_V3:
+            self.assertIs(type(CM[int, str]()), CM)
+
+    def test_chainmap_subclass(self):
+
+        class MyChainMap(typing_extensions.ChainMap[str, int]):
+            pass
+
+        cm = MyChainMap()
+        self.assertIsInstance(cm, MyChainMap)
+
+        self.assertIsSubclass(MyChainMap, collections.ChainMap)
+        if TYPING_V3:
+            self.assertNotIsSubclass(collections.ChainMap, MyChainMap)
+
+    def test_deque_instantiation(self):
+        self.assertIs(type(typing_extensions.Deque()), collections.deque)
+        self.assertIs(type(typing_extensions.Deque[T]()), collections.deque)
+        self.assertIs(type(typing_extensions.Deque[int]()), collections.deque)
+        class D(typing_extensions.Deque[T]): ...
+        if TYPING_V3:
+            self.assertIs(type(D[int]()), D)
```

Comment:
Potentially you might also add a test for ``__bases__`` they are also type-erased since one of the releases of ``typing``.

(But choose a class that actually derives from something subscripted, maybe ``Counter``.)